components/experimental/src/measure/measureunit.rs

```diff
@@ -62,160 +61,90 @@ impl MeasureUnit {
+    /// assert_eq!(short_representation, "P2I85", "{}", "square-meter");
     ///
-    /// let full_unit = "meter-per-second";
-    /// let measure_unit = parser.try_from_str(full_unit).unwrap();
-    /// let short_representation = measure_unit.generate_short_representation();
-    /// assert_eq!(short_representation, "I85#R#P-1I127", "{}", full_unit);
-    /// assert_eq!(full_unit.len() - short_representation.len(), 3, "{}", full_unit);
-    ///
-    /// let full_unit = "liter-per-100-kilometer";
-    /// let measure_unit = parser.try_from_str(full_unit).unwrap();
-    /// let short_representation = measure_unit.generate_short_representation();
-    /// assert_eq!(short_representation, "I82#R#C100#P-1D3I85", "{}", full_unit);
-    /// assert_eq!(full_unit.len() - short_representation.len(), 4, "{}", full_unit);
-    ///
-    /// let full_unit = "portion-per-1e9";
-    /// let measure_unit = parser.try_from_str(full_unit).unwrap();
-    /// let short_representation = measure_unit.generate_short_representation();
-    /// assert_eq!(short_representation, "I113#R#C1E9", "{}", full_unit);
-    /// assert_eq!(full_unit.len() - short_representation.len(), 4, "{}", full_unit);
-    ///
-    /// let full_unit = "per-10000000000-portion";
-    /// let measure_unit = parser.try_from_str(full_unit).unwrap();
-    /// let short_representation = measure_unit.generate_short_representation();
-    /// assert_eq!(short_representation, "R#C1E10#P-1I113", "{}", full_unit);
-    /// assert_eq!(full_unit.len() - short_representation.len(), 8, "{}", full_unit);
     ///
-    /// let full_unit = "liter-per-240000000000-kilometer";
-    /// let measure_unit = parser.try_from_str(full_unit).unwrap();
+    /// let measure_unit = parser.try_from_str("liter-per-100-kilometer").unwrap();
     /// let short_representation = measure_unit.generate_short_representation();
-    /// assert_eq!(short_representation, "I82#R#C24E10#P-1D3I85", "{}", full_unit);
-    /// assert_eq!(full_unit.len() - short_representation.len(), 11, "{}", full_unit);
-    ///
-    /// let full_unit = "millimeter-per-square-microsecond";
-    /// let measure_unit = parser.try_from_str(full_unit).unwrap();
-    /// let short_representation = measure_unit.generate_short_representation();
-    /// assert_eq!(short_representation, "D-3I85#R#P-2D-6I127", "{}", full_unit);
-    /// assert_eq!(full_unit.len() - short_representation.len(), 14, "{}", full_unit);
-    ///
+    /// assert_eq!(short_representation, "C100I82P-1D3I85", "{}", "liter-per-100-kilometer");
     /// ```
     pub fn generate_short_representation(&self) -> String {
         // Convert the constant to scientific notation if it is a power of 10 with more than 3 trailing zeros
-        fn power_of_10_to_scientific(n: u64) -> String {
+        fn append_power_of_10_to_scientific(n: u64, append_to: &mut String) {
             if n < 1000 {
-                return n.to_string();
+                append_to.push_str(&n.to_string());
```

Comment:
this allocates an intermediate string.
```suggestion
                let _infallible = write!(append_to, "{n}");
```